### PR TITLE
Fix minor issues from linters

### PR DESCRIPTION
**Pull request by hypnoglow**
I've run some linters against the code, which revealed a few issues. This PR fixes them.

### CODE REVIEW

File: cmd/git-sync/main.go [hunk 1]
```diff
@@ -353,7 +354,7 @@ func setFlagDefaults() {
 // it is deadlocked.
 func sleepForever() {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
```
Comment (hypnoglow) [inline cmd/git-sync/main.go]: SIGKILL cannot be trapped

File: cmd/git-sync/main.go [hunk 2]
```diff
@@ -527,7 +528,7 @@ func revIsHash(ctx context.Context, rev, gitRoot string) (bool, error) {
 func syncRepo(ctx context.Context, repo, branch, rev string, depth int, gitRoot, dest string) (bool, error) {
 	target := path.Join(gitRoot, dest)
 	gitRepoPath := path.Join(target, ".git")
-	hash := rev
```
Comment (hypnoglow) [inline cmd/git-sync/main.go]: rationale: `switch` has `default` case where hash variable is assigned, so this assignement has no effect.

File: cmd/git-sync/main.go [hunk 3]
```diff
@@ -659,7 +660,7 @@ func setupGitSSH(setupKnownHosts bool) error {
 	}
 
 	if setupKnownHosts {
-		_, err := os.Stat(pathToSSHKnownHosts)
+		_, err = os.Stat(pathToSSHKnownHosts)
```
Comment (hypnoglow) [inline cmd/git-sync/main.go]: rationale: `err` assignment on 668 affects only `if` scope so error check on 674 won't see the error if it occurs.

File: cmd/git-sync/main.go (hunk 1 above)
Comment (thockin) [reply cmd/git-sync/main.go]: That was the point - it's not actually trying to do anything, just trick Go's runtime into thinking it is not hung.  This leaves "real" signals to be handled elsewhere if needed, which seems plausible for SIGINT.

I propose we nix this - I am thinking about changing this functionality anyway, so it won't matter long term.

File: cmd/git-sync/main.go (hunk 3 above)
Comment (thockin) [reply cmd/git-sync/main.go]: good one.

File: cmd/git-sync/main.go (hunk 2 above)
Comment (thockin) [reply cmd/git-sync/main.go]: good one

File: cmd/git-sync/main.go (hunk 1 above)
Comment (hypnoglow) [reply cmd/git-sync/main.go]: Ok, thanks for the explanation. Reverted.

**Pull request merged**